Todo edit feature

valeshgopal opened the pull request:
> First thing, PLEASE READ THIS: [ReactPlay Code Review Checklist](https://github.com/reactplay/react-play/wiki/ReactPlay-Code-Review-Checklist)

# Description

Added 'edit task' feature to the 'To-do App'. Users can correct any typos or add extra info with the help of this functionality.

Fixes # (issue)
Previously there was no option to edit a task.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)


# How Has This Been Tested?

- Click on the 'edit' icon
- Make changes to the task
- Submit the task
- Task is updated

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

Programming-School-Pro-Coding reviewed (commented):
Hi Sir

I suggest to use LocalStorage to save the data to the browser

If you want to more explanation tell me

atapas commented:
@valeshgopal Thanks for the play. Could you please create an issue for this play?

Programming-School-Pro-Coding reviewed (dismissed):
LGTM

valeshgopal commented:
> @valeshgopal Thanks for the play. Could you please create an issue for this play?

Hi Tapas! I've opened the issue.

atapas commented on src/plays/react-todo-app/ReactTodoApp.jsx:
```diff
@@ -11,50 +11,81 @@ import './react-todo.css';
 import Input from './components/Input';
 import TodoList from './components/TodoList';
 
+function getLocalStorageTodos() {
```
I see an existing hook to handle localstorage operation: https://github.com/reactplay/react-play/blob/main/src/plays/keeper/Hooks/useLocalStorage.js

But this hook is unfortunately inside a play. Could you please move the hook here `https://github.com/reactplay/react-play/tree/main/src/common/hooks` and then use it?

It is better we use reusable code across. Please modify the import of the existing play where it is being used once you move it.

atapas commented on src/plays/react-todo-app/ReactTodoApp.jsx:
```diff
@@ -11,50 +11,81 @@ import './react-todo.css';
 import TodoList from './components/TodoList';
 
+function getLocalStorageTodos() {
+  const todos = localStorage.getItem('todos');
```
Always use keys that are unique. May be another play have same key name and override yours.

valeshgopal replied:
> Always use keys that are unique. May be another play have same key name and override yours.

Added folder name before localstorage key to make it more unique - (eg: useLocalStorage('react-todo-app/todos', []), useLocalStorage('keeper/notes', []))

atapas reviewed (changes requested):
@valeshgopal

I've requested a couple of refactoring to make the code organization better. Could you please take a look.

Also the UI breaks in the responsive mode, and need a fix. If you could check this as well, will be great.

![image](https://user-images.githubusercontent.com/3633137/171445764-f14e18e4-5fe4-4d43-8fa5-2bcbc00868d5.png)

valeshgopal commented:
Hi Tapas! I've made the following changes:
- Changed UI look to make it feel more aesthetic 
- Made the form responsive
- Moved local Storage hook from https://github.com/reactplay/react-play/blob/main/src/plays/keeper/Hooks/useLocalStorage.js to https://github.com/reactplay/react-play/tree/main/src/common/hooks and imported it from there.
- Added folder name before localstorage key to make it more unique - (eg: useLocalStorage('react-todo-app/todos', []), useLocalStorage('keeper/notes', []))

atapas commented on src/plays/keeper/Keeper.jsx:
```diff
@@ -1,23 +1,23 @@
 import CreateArea from './CreateArea';
 import Note from './Notes';
-import useLocalStorage from './Hooks/useLocalStorage';
+import useLocalStorage from '../../common/hooks/useLocalStorage';
```
The import should work like this too?

import useLocalStorage from 'common/hooks/useLocalStorage';

valeshgopal replied:
I tried that path at first. But it gives the following error message: 
Module not found: Error: You attempted to import /common/hooks/useLocalStorage which falls outside of the project src/ directory. Relative imports outside of src/ are not supported.
You can either move it inside src/, or add a symlink to it from project's node_modules/.

atapas replied:
Interesting: See this usage. We are doing already

https://github.com/reactplay/react-play/blob/34dbcb1b6d8035b8813e92dfdcce6f5b83215cca/src/plays/react-todo-app/ReactTodoApp.jsx#L3

atapas replied:
Should you add hooks here?

https://github.com/reactplay/react-play/blob/main/src/common/index.js

atapas commented:
> Hi Tapas! I've made the following changes:
> 
> * Changed UI look to make it feel more aesthetic
> * Made the form responsive
> * Moved local Storage hook from https://github.com/reactplay/react-play/blob/main/src/plays/keeper/Hooks/useLocalStorage.js to https://github.com/reactplay/react-play/tree/main/src/common/hooks and imported it from there.
> * Added folder name before localstorage key to make it more unique - (eg: useLocalStorage('react-todo-app/todos', []), useLocalStorage('keeper/notes', []))

I loved that. Thank You!

Given a tiny comment. Now I will test the changes from UI too.

atapas commented:
@all-contributors please add @valeshgopal for Code
